src/futils.c

```diff
@@ -24,30 +24,44 @@ int git_futils_mkpath2file(const char *file_path, const mode_t mode)
 
 int git_futils_mktmp(git_buf *path_out, const char *filename, mode_t mode)
 {
-	int fd;
-	mode_t mask;
-
-	p_umask(mask = p_umask(0));
+	int fd, count;
+	char *template;
+	uint64_t random;
+	const char *alphabet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890_-";
 
 	git_buf_sets(path_out, filename);
 	git_buf_puts(path_out, "_git2_XXXXXX");
 
 	if (git_buf_oom(path_out))
 		return -1;
 
-	if ((fd = p_mkstemp(path_out->ptr)) < 0) {
-		git_error_set(GIT_ERROR_OS,
-			"failed to create temporary file '%s'", path_out->ptr);
-		return -1;
-	}
+	// This avoids a call to `mkstemp(3)`, because that way there is no need to
+	// call `chmod(2)`, which causes issues under some platforms (e.g. with NFS).
```

Comment:
```suggestion
	/*
	 * This avoids a call to `mkstemp(3)`, because that way there is no need to
	 * call `chmod(2)`, which causes issues under some platforms (e.g. with NFS).
	 */
```

oh crud, this is totally my fault. I forgot that we use block comments in this project. 🙇

Reply:
Well, I could've taken the time to look at the codebase myself instead of blindly copying your patch :P